components/autocomplete/index.js

```diff
@@ -0,0 +1,268 @@
+		let pattern = '^';
+
+		if ( triggerPrefix ) {
+			// Trigger prefix may contain reserved characters of a regular
+			// expression, so escape
+			pattern += escapeStringRegexp( triggerPrefix );
+		}
+
+		// Create matching group for the search value
+		pattern += '([^\\s]*)$';
+
+		pattern = new RegExp( pattern );
+
+		let match, search;
+		if ( ( match = value.match( pattern ) ) ) {
+			search = new RegExp( escapeStringRegexp( match[ 1 ] ), 'i' );
+		}
+
+		if ( isOpen ) {
+			if ( match ) {
+				// Reset selection to initial when search changes
+				this.setState( {
+					position: this.getInputPosition(),
+					selectedIndex: 0,
+					search,
+				} );
+			} else {
+				// This may occur when pressing space, for example
+				this.reset();
+			}
+		} else if ( match ) {
+			this.setState( {
+				isOpen: true,
+				position: this.getInputPosition(),
+				search,
+			} );
+		}
+	}
+
+	getFilteredOptions() {
+		const { options, maxResults = 10 } = this.props;
+		const { search } = this.state;
+
+		const filtered = [];
+		for ( let i = 0; i < options.length; i++ ) {
+			const option = options[ i ];
+
+			// Merge label into keywords
+			let { keywords = [] } = option;
+			if ( 'string' === typeof option.label ) {
+				keywords = [ ...keywords, option.label ];
+			}
+
+			const isMatch = keywords.some( ( keyword ) => search.test( keyword ) );
+			if ( ! isMatch ) {
+				continue;
+			}
+
+			filtered.push( option );
+
+			// Abort early if max reached
+			if ( filtered.length === maxResults ) {
+				break;
+			}
+		}
+
+		return filtered;
+	}
+
+	setSelectedIndex( event ) {
+		const { isOpen, selectedIndex } = this.state;
+		if ( ! isOpen ) {
+			return;
+		}
+
+		const options = this.getFilteredOptions();
+
+		let nextSelectedIndex;
+		switch ( event.keyCode ) {
+			case UP:
+				nextSelectedIndex = ( selectedIndex === 0 ? options.length : selectedIndex ) - 1;
+				this.setState( { selectedIndex: nextSelectedIndex } );
+				break;
+
+			case DOWN:
+				nextSelectedIndex = ( selectedIndex + 1 ) % options.length;
+				this.setState( { selectedIndex: nextSelectedIndex } );
+				break;
+
+			case ESCAPE:
+				this.reset();
+				break;
+
+			case ENTER:
+				this.select( options[ selectedIndex ] );
+				break;
+
+			default:
+				return;
+		}
+
+		// Any handled keycode should prevent original behavior. This relies on
+		// the early return in the default case.
+		event.preventDefault();
+		event.stopImmediatePropagation();
+	}
+
+	render() {
+		const { children, className } = this.props;
+		const { position, isOpen, selectedIndex } = this.state;
+		const classes = classnames( 'components-autocomplete', className );
+
+		// Blur is applied to the wrapper node, since if the child is Editable,
+		// the event will not have `relatedTarget` assigned.
+		return (
+			<div
+				ref={ this.bindNode }
+				onBlur={ this.onBlur }
+				className={ classes }
+			>
+				{ cloneElement( Children.only( children ), {
```

Comment:
IIRC, the input (or conteneditable) should have the following aria attributes:


```
role="combobox"
aria-expanded={ isExpanded }
aria-autocomplete="list"
aria-owns={ suggestionsListID }
aria-activedescendant={ selectedSuggestionListItemId }
```
the list should have

```
role="listbox"
```

and each list item should have

```
role="option"
aria-selected={ index === this.props.selectedIndex }
```

Could mimic the `FormTokenField` component